File: frontend/types/medication.ts
export interface ActiveIngredient {
  id: number;
  name_en: string;
  name_ar: string;
}

export interface Medication {
  id: number;
  name_en: string;
  name_ar: string;
  dosage_form: string | null;
  strength: string | null;
  manufacturer: string | null;
  requires_prescription: boolean;
  price: string;
  general_usage: string | null;
  general_warnings: string | null;
  pregnancy_warning: string | null;
  pediatric_warning: string | null;
  elderly_warning: string | null;
  active_ingredient: ActiveIngredient | null;
}

export interface MedicationSearchResult {
  medication: Medication;
  in_stock: boolean;
  quantity_available: number;
  substitutes: Medication[];
}

export interface MedicationCreatePayload {
  name_en: string;
  name_ar: string;
  active_ingredient_id?: number | null;
  dosage_form?: string;
  strength?: string;
  manufacturer?: string;
  requires_prescription?: boolean;
  price: number;
  general_usage?: string;
  general_warnings?: string;
}

export interface InventoryItem {
  id: number;
  medication_id: number;
  quantity: number;
  reorder_threshold: number;
  batch_no: string | null;
  expiry_date: string | null;
  is_low_stock: boolean;
}

export interface InventoryUpdatePayload {
  quantity?: number;
  reorder_threshold?: number;
  batch_no?: string;
  expiry_date?: string;
}

File: backend/app/schemas/medication.py
from datetime import datetime
from decimal import Decimal
from pydantic import BaseModel


class ActiveIngredientOut(BaseModel):
    id: int
    name_en: str
    name_ar: str

    class Config:
        from_attributes = True


class MedicationOut(BaseModel):
    id: int
    name_en: str
    name_ar: str
    dosage_form: str | None
    strength: str | None
    manufacturer: str | None
    requires_prescription: bool
    price: Decimal
    general_usage: str | None
    general_warnings: str | None
    pregnancy_warning: str | None
    pediatric_warning: str | None
    elderly_warning: str | None
    active_ingredient: ActiveIngredientOut | None = None

    class Config:
        from_attributes = True


class MedicationSearchResult(BaseModel):
    medication: MedicationOut
    in_stock: bool
    quantity_available: int
    substitutes: list[MedicationOut] = []


class MedicationCreate(BaseModel):
    name_en: str
    name_ar: str
    active_ingredient_id: int | None = None
    dosage_form: str | None = None
    strength: str | None = None
    manufacturer: str | None = None
    requires_prescription: bool = True
    price: Decimal = Decimal("0")
    general_usage: str | None = None
    general_warnings: str | None = None
    pregnancy_warning: str | None = None
    pediatric_warning: str | None = None
    elderly_warning: str | None = None


class InventoryItemOut(BaseModel):
    id: int
    medication_id: int
    quantity: int
    reorder_threshold: int
    batch_no: str | None
    expiry_date: datetime | None
    is_low_stock: bool

    class Config:
        from_attributes = True


class InventoryUpdate(BaseModel):
    quantity: int | None = None
    reorder_threshold: int | None = None
    batch_no: str | None = None
    expiry_date: datetime | None = None
